pkg/gornir/runner.go

```diff
@@ -5,69 +5,39 @@ import (
 	"sync"
 )
 
-type JobParameters struct {
-	title  string
-	logger Logger
-	host   *Host
-}
-
-func NewJobParameters(title string, logger Logger) *JobParameters {
-	return &JobParameters{
-		title:  title,
-		logger: logger,
-	}
-}
-
-func (jp *JobParameters) Title() string {
-	return jp.title
-}
-
-func (jp *JobParameters) Logger() Logger {
-	return jp.logger
-}
-
-func (jp *JobParameters) Host() *Host {
-	return jp.host
-}
-
-func (jp *JobParameters) ForHost(host *Host) *JobParameters {
-	return &JobParameters{
-		title:  jp.title,
-		logger: jp.logger.WithField("host", host.Hostname),
-		host:   host,
-	}
-}
+// TaskInstanceResult is the resut of running a task on a given host
+type TaskInstanceResult interface{}
 
 // Task is the interface that task plugins need to implement.
 // the task is responsible to indicate its completion
 // by calling sync.WaitGroup.Done()
 type Task interface {
-	Run(context.Context, *sync.WaitGroup, *JobParameters, chan *JobResult)
+	Run(context.Context, Logger, *Host) (TaskInstanceResult, error)
```

Comment:
I think we could change the name of `Run` either here or at the Runner to make the workflow easier to follow. So when I see a Run, I immediately know if we are talking about a Task OR a Runner (I know, they have different signatures, but still)

So, move from "The Runner **Run** the Task that is **Run**" to "The Runner **Run** the Task that it **Execute**" or "The Runner **Trigger** the Task that is **Run**" or something else.